### PR TITLE
Do not inherit kw font size if it was not used due to MathML

### PR DESCRIPTION
**Pull request by Manishearth**
See https://bugzilla.mozilla.org/show_bug.cgi?id=1361126#c9

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16695)

### CODE REVIEW

**Comment by Manishearth** [comment]
r? @heycam 

(emilio isn't comfortable with reviewing this)

File: components/style/properties/longhand/font.mako.rs [hunk 1]
```diff
@@ -927,10 +927,14 @@ ${helpers.single_keyword_system("font-variant-caps",
...
         });
-        context.mutate_style().mutate_font()
+        let used_kw = context.mutate_style().mutate_font()
                .inherit_font_size_from(parent, kw_inherited_size);
-        context.mutate_style().font_size_keyword =
-            context.inherited_style.font_size_keyword;
+        if used_kw {
+            context.mutate_style().font_size_keyword =
+                context.inherited_style.font_size_keyword;
+            } else {
```
Comment (emilio) [inline components/style/properties/longhand/font.mako.rs]: nit: Indentation is off here.

**Review by emilio — COMMENTED** [review]
This makes sense conceptually to me, but I'd have to spend a lot of time to get how all the pieces fit together, so if Cam could review it, that'd be neat :)

File: components/style/properties/gecko.mako.rs [hunk 2]
```diff
@@ -1490,12 +1490,15 @@ fn static_assert() {
...
+    /// Returns true if the inherited size was actually used
     pub fn inherit_font_size_from(&mut self, parent: &Self,
-                                  kw_inherited_size: Option<Au>) {
+                                  kw_inherited_size: Option<Au>) -> bool {
         let (adjusted_size, adjusted_unconstrained_size)
             = self.calculate_script_level_size(parent);
         if adjusted_size.0 != parent.gecko.mSize ||
-           adjusted_unconstrained_size.0 != parent.gecko.mScriptUnconstrainedSize {
+           adjusted_unconstrained_size.0 != parent.gecko.mScriptUnconstrainedSize ||
+           parent.gecko.mSize != parent.gecko.mScriptUnconstrainedSize {
```
Comment (heycam) [inline components/style/properties/gecko.mako.rs]: Can you explain what the addition of this third condition is for?  I can't quite work it out.

Comment (Manishearth) [inline components/style/properties/gecko.mako.rs]: Hm, on thinking about this it's actually unnecessary. Removing.

File: components/style/properties/gecko.mako.rs [hunk 3]
```diff
@@ -1490,8 +1490,10 @@ fn static_assert() {
     /// This function will also handle scriptminsize and scriptlevel
     /// so should not be called when you just want the font sizes to be copied.
     /// Hence the different name.
+    ///
+    /// Returns true if the inherited size was actually used
```
Comment (heycam) [inline components/style/properties/gecko.mako.rs]: Should probably clarify that you mean the inherited keyword-based size here (and not say the sizes in `parent`).

**Review by heycam — APPROVED** [review]

File: components/style/properties/gecko.mako.rs (hunk 3 above)
Comment (Manishearth) [inline components/style/properties/gecko.mako.rs]: fixed

**Pull request merged**